astropy/coordinates/representation/base.py

```diff
@@ -147,24 +147,33 @@ class BaseRepresentationOrDifferential(MaskableShapedLikeNDArray):
     # object arrays.
     __array_priority__: Final = 50000
 
-    name: ClassVar[str]
+    # Have to define this default b/c init_subclass is not called for the base class.
+    name: ClassVar[str] = "base"
     """Name of the representation or differential.
 
-    By default, the lower-cased name of the class with with any trailing
-    'representation' or 'differential' removed. (E.g., 'spherical' for
-    `~astropy.coordinates.SphericalRepresentation` or
+    When a subclass is defined, by default, the name is the lower-cased name of the
+    class with with any trailing 'representation' or 'differential' removed. (E.g.,
+    'spherical' for `~astropy.coordinates.SphericalRepresentation` or
     `~astropy.coordinates.SphericalDifferential`.)
+
+    This can be customized when defining a subclass by setting the class attribute.
     """
 
     info = BaseRepresentationOrDifferentialInfo()
 
     def __init_subclass__(cls) -> None:
-        # Name of the representation or differential.
-        cls.name = (
+        # Name of the representation or differential
+        default_name = (
             cls.__name__.lower()
             .removesuffix("representation")
             .removesuffix("differential")
         )
+        if not hasattr(cls, "name"):
+            cls.name = default_name
+        elif BaseRepresentationOrDifferential not in cls.__bases__ and cls.name in (
```

Comment:
Looking at this again, I'm actually not sure why this first part is in here - just because it doesn't have `.name` itself? But then wouldn't that be caught? I.e., why not just have,
```
# Set default name if no name was given or we inherited it
if (name := getattr(cls, "name", None)) is None or name in (getattr(base, "name", None) for base in cls.__bases__):
   cls.name = cls.__name__.lower()...
```

Actually, can we not just do `if "name" not in cls.__dict__`?

Reply:
I think we can.

Reply:
Done.